### PR TITLE
feat: resource scan

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx [hunk 1]
```diff
@@ -19,10 +19,63 @@ import AppDetailsStore from '../../appDetails.store'
...
 import { ReactComponent as Trash } from '../../../../../assets/icons/ic-delete-interactive.svg'
-import { importComponentFromFELibrary } from '../../../../common'
+import { getShowResourceScanModal, importComponentFromFELibrary } from '../../../../common'
+import { createBody } from '../nodeDetail/nodeDetail.api'
 
+const OpenVulnerabilityModalButton = importComponentFromFELibrary('OpenVulnerabilityModalButton', null, 'function')
+const ScanResourceModal = importComponentFromFELibrary('ScanResourceModal', null, 'function')
 const DeploymentWindowConfirmationDialog = importComponentFromFELibrary('DeploymentWindowConfirmationDialog')
 
+const PodPopup: React.FC<{
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx:29]: Extract it out to separate component

File: src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx [hunk 2]
```diff
@@ -102,6 +120,18 @@ export default function ResourceBrowserActionMenu({
                     removeTabByIdentifier={removeTabByIdentifier}
                 />
             )}
+
+            {showVulnerabilityModal && ScanResourceModal && (
+                <ScanResourceModal
```
Comment (eshankvaish) [inline src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx:125]: pls check if both the modals won't open together

**Review by shivani170 — DISMISSED** [review]

File: src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx [hunk 3]
```diff
@@ -173,9 +141,33 @@ const NodeDeleteComponent = ({
...
                 </PopupMenu.Button>
                 <PopupMenu.Body>
-                    <PodPopup kind={nodeDetails?.kind} describeNode={describeNodeWrapper} />
+                    <PodPopup
+                        kind={nodeDetails?.kind}
+                        describeNode={describeNodeWrapper}
+                        toggleShowDeleteConfirmation={toggleShowDeleteConfirmation}
+                        isExternalArgoApp={isExternalArgoApp}
+                        handleShowVulnerabilityModal={handleShowVulnerabilityModal}
+
```
Comment (eshankvaish) [inline src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx:150]:
```suggestion
```

File: src/components/v2/appDetails/k8Resource/nodeType/PodPopup.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        describeNode(NodeDetailTabs.EVENTS)
+    }
+
+    const handleDescribeLogs = () => {
+        describeNode(NodeDetailTabs.LOGS)
+    }
+
+    return (
+        <div className="pod-info__popup-container">
+            {kind === NodeType.Pod ? (
```
Comment (eshankvaish) [inline src/components/v2/appDetails/k8Resource/nodeType/PodPopup.tsx:30]: use &&

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/v2/appDetails/k8Resource/nodeType/NodeDelete.component.tsx]: Done

File: src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/ResourceBrowser/ResourceList/ResourceBrowserActionMenu.tsx:125]: Checked

File: src/components/v2/appDetails/k8Resource/nodeType/PodPopup.tsx (hunk 4 above)
Comment (AbhishekA1509) [reply src/components/v2/appDetails/k8Resource/nodeType/PodPopup.tsx:30]: Done

**Review by eshankvaish — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**